tools/cmake/utility.cmake

```diff
@@ -0,0 +1,68 @@
+function(set_macro_definitions)
```

Comment:
This is the meat of the PR

Reply:
Also, lemme know if you can think of a better function name for this.

Reply:
No objections to the name -- can't really think of a better way to describe this function.